crates/mysten-network/src/multiaddr.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use multiaddr::{Multiaddr, Protocol};
+use std::{
+    borrow::Cow,
+    net::{IpAddr, SocketAddr},
+};
+
+pub(crate) fn parse_tcp<'a, T: Iterator<Item = Protocol<'a>>>(protocols: &mut T) -> Result<u16> {
```

Comment:
Nit: an `impl Iterator<Item = ...>` might lighten the type signature on these `parse_` functions.